### ISSUE TITLE
CSS isolation doesn't seem to work with lazy loaded assembly

### ISSUE THREAD

**Issue by sm15455** [dotnet/aspnetcore#26542]
Hi,

I have a solution containing 2 projects: a Blazor RC1 project and library project.
The library contains a single component and is lazy loaded by the application. The component has a css file that follows css isolation naming rules. When I load the assembly, the assembly is correctly loaded and the component rendered, but the CSS file is ignored. The CSS for the component is not included in the scoped.style.css file (I think it's correct), but it's not even loaded at run time. Is it a bug or this scenario is simply not supported?

**Comment by pranavkm** [comment]
FYI @captainsafia

**Comment by captainsafia** [comment]
@sm15455 Can you share a repro project? I was not able to repro this issue on 5.0.0-rc.2.20465.7 (note: RC2). Also, I assume you are using the lazy-loading feature as documented [here](https://docs.microsoft.com/en-us/aspnet/core/blazor/webassembly-lazy-load-assemblies?view=aspnetcore-5.0).

FWIW, with the way that lazy-loading is setup, the isolated CSS is loaded when the application is launched. The CSS itself is applied when the component is instantiated. So the flow looks roughly like this.

1. User navigates to page. Isolated component's scoped CSS file is loaded.
2. User navigates to route with lazy-loaded assembly. Isolated component's scoped CSS is applied. Any resources referenced in the CSS are fetched.

**Comment by sm15455** [comment]
@captainsafia I have investigate the issue and have found my problem description to be misleading. Basically, everything works but a little exception. here is what I did.

1. Created a new blazor project (RC1)
2. Create a library project
3. Moved the counter.razor file into the library
4. Configured the project to lazy load the library (I used the official doc you linked)
5. Pressed F5

Now the browser opens and everything is ok, the component is loaded. I close the browser and.

6. Create a counter.razor.css file and add a css style to it.
7. Pressed F5

Now the browser opens, and the component is loaded but the style is not applied because it's not included in the scoped.style.css file.
I found out that to make it work, I have to rebuild the entire solution after any change to the css file and then press F5.
